server/src/plugins/tx-monitor.ts

```diff
@@ -0,0 +1,69 @@
+import fp from 'fastify-plugin';
+import { FastifyInstance } from 'fastify';
+import { IncomingMessage, ServerResponse, Server } from 'http';
+
+declare module 'fastify' {
+  export interface FastifyInstance<
+    HttpServer = Server,
+    HttpRequest = IncomingMessage,
+    HttpResponse = ServerResponse
+  > {
+    authenticate: any;
+  }
+}
+
+const cron = require('node-cron');
+const fetch = require('node-fetch');
+
+import { getPendingTxs, updateTransactionStatus } from '../db';
+import { TransactionStatus } from '../types';
+
+const EXPLORER_API_BASE = 'https://api.blockcypher.com/v1';
+
+export default fp(function transactionsMonitorCron(
+  fastify: FastifyInstance<Server, IncomingMessage, ServerResponse>,
+  opts,
+  next
+) {
+    const monitor = async () => {
+        const pendingTransactions = await getPendingTxs();
+        if (!pendingTransactions || pendingTransactions.length === 0) return;
+        ​
+        const txPromises = pendingTransactions.map(async ({ tx_hash: txHash }) => {
+            const json: any = await getTxStatus(txHash);
+        ​
+            if (json) {
+                if (json.confirmed) {
+                    await updateTransactionStatus(txHash, TransactionStatus.passed);
+                } else if (json.execution_error) {
+                    await updateTransactionStatus(txHash, TransactionStatus.failed);
+                }
+            }
+        ​
+            return json;
+        });
+        ​
+        const results = Promise.all(txPromises);
+        return results;
+    };
+      ​
+
+  fastify.decorate('updateTransactions', async () => {
+    cron.schedule('*/12 * * * * *', () => monitor());
+  });
+
+  // @ts-ignore
+  fastify.updateTransactions();
+
+  next();
+});
+
+async function getTxStatus<A>(hash: string): Promise<A> {
+  const url = `${EXPLORER_API_BASE}/eth/main/txs/${encodeURIComponent(hash)}`;
```

Comment:
Wasn't an API KEY required?

Reply:
No, all GET request are FREE and they don't require the use of an API key

Reply:
I guess we're good hitting the API every 12 minutes and without an API key.

```markdown
## Rate Limits
BlockCypher APIs can be used up to these rate limits:

Classic requests, up to 5 requests/sec and 600 requests/hr
WebHooks and WebSockets, up to 600 requests/hr
If you exceed a limit, an HTTP status code 429 will be returned. Please register for a user token if your usage can exceed those limits. Our future pricing plan will be tiered based on usage and volume. We will have a free tier and an extended free plan for our early users. To request higher limits or SLAs, please email us.
```